packages/core/src/types.ts

```diff
@@ -478,3 +478,28 @@ export interface InternalOptions<TProviderType = ProviderType> {
   cookies: CookiesOptions
   callbackUrl: string
 }
+
+/**
+ * Util type that matches some strings literally, but allows any other string as well.
+ * @source https://github.com/microsoft/TypeScript/issues/29729#issuecomment-832522611
+ */
+export type LiteralUnion<T extends U, U = string> =
+  | T
+  | (U & Record<never, never>)
+
+export interface SignInOptions extends Record<string, unknown> {
+  returnTo?: string
+  redirect?: boolean
+}
+
+/** Match `inputType` of `new URLSearchParams(inputType)` */
+export type SignInAuthorizationParams =
+  | string
+  | string[][]
+  | Record<string, string>
+  | URLSearchParams
```

Comment:
```suggestion
```

This can just be `Parameters<URLSearchParams>[0]` where you need it.

Reply:
it can't because it is a class.
but it can be something like:

```ts
export type SignInAuthorizationParams = typeof URLSearchParams extends new (
  ...args: infer U
) => URLSearchParams
  ? U[0]
  : never
  ```
  
  i prefer this because it is not hardcoded but gets the current type